File: updater/src/main/java/updater/utility/FieldWrapper.java
package updater.utility;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import org.objectweb.asm.Type;
import org.objectweb.asm.tree.FieldNode;
import updater.Updater;

import java.lang.reflect.Modifier;

@RequiredArgsConstructor
@Getter
@Setter
public class FieldWrapper {
    private final FieldNode fieldNode;

    public boolean hasDimensions(int i) {
        return Type.getType(fieldNode.desc).getDimensions() == i;
    }

    public boolean isOfType(Type type) {
        return Type.getType(fieldNode.desc).equals(type);
    }

    public Type getType() {
        return Type.getType(fieldNode.desc);
    }

    public boolean typeInJar() {
        return Updater.classMap.values().stream().anyMatch(classWrapper -> getType().getElementType().equals(Type.getObjectType(classWrapper.getClassNode().name)));
    }

    public boolean isStatic() {
        return Modifier.isStatic(fieldNode.access);
    }
}

File: find-names/src/main/java/mapper/wrappers/FieldWrapper.kt
package mapper.wrappers

import org.objectweb.asm.Type
import org.objectweb.asm.tree.FieldNode

class FieldWrapper(val jar: JarWrapper, val klass: ClassWrapper, val node: FieldNode) {

    val name: String get() = node.name

    val access get() = node.access

    val desc: String get() = node.desc

    val type: Type = Type.getType(desc)

    val id = klass.type to name

    override fun toString(): String {
        return "$klass.$name"
    }
}